src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -157,6 +158,9 @@ def exception_handler(exception: Exception, execution: ExecutionMetadata):
 
         log.debug("Executing command exception callback")
         log_command_execution_error(exception, execution)
+        if isinstance(exception, DatabaseError):
+            return ClickException(exception.msg)
```

Comment:
Telemetry will then report this as a non-CLI error too, right?

Any thoughts on whether we should do something similar for ProgrammingError coming out of the connector?

Reply:
> Telemetry will then report this as a non-CLI error too, right?

That's correct, reporting happens before type casting. I think it's ok. We don't distinguish between expected and unexpected db errors.

`ProgrammingError` is a `DatabaseError ` so it's already covered.